src/file_object.cpp

```diff
@@ -7,6 +7,16 @@
 
 namespace Natalie {
 
+// If it's not a string but has a to_path method then execute that method.
+// make sure the path or to_path result is a String before continuing.
+// this is common to many functions probably belongs somewhere else
+Value ConvertToPath(Env *env, Value path) {
```

Comment:
It wasnt clear to me where to put this helper method, so i put it here, but I think some of the `Dir` methods would also want to use it, so maybe it should be moved to natalie.hpp/natalie.cpp ?